### PR TITLE
Consistent bool returns on exposed methods for err handling

### PR DESCRIPTION
**Pull request by eparis**
This change is [Reviewable](https://reviewable.kubernetes.io/reviews/kubernetes/contrib/1645)

### CODE REVIEW

**Comment by lavalamp** [comment]
I recall debating this in another PR, did you convince past me that this was better than returning errors?

**Comment by eparis** [comment]
'its how everything else is' was my argument and 1/2 1/2 is worse than all one or all the other.  Errorf() at the bottom your munger code just bails.

We can switch everything to errors in another go if we have some consistency. (and this fixes real bugs).

**Comment by lavalamp** [comment]
OK, sure. @foxish can you do a careful review? I only skimmed :)

**Comment by foxish** [comment]
@lavalamp ack. I'll get to it later today.

File: mungegithub/github/github.go [hunk 1]
```diff
@@ -558,35 +559,39 @@ func (config *Config) NewIssue(title, body string, labels []string, owner string
...
-	return ""
+	return "", ok
 }
 
 // IsForBranch return true if the object is a PR for a branch with the given
 // name. It return false if it is not a pr, it isn't against the given branch,
 // or we can't tell
-func (obj *MungeObject) IsForBranch(branch string) bool {
-	objBranch := obj.Branch()
+func (obj *MungeObject) IsForBranch(branch string) (bool, bool) {
```
Comment (foxish) [inline mungegithub/github/github.go]: I feel that to return (bool, bool) is confusing, in that one needs to check the result of the second return statement to know if the first is meaningful.

**Comment by foxish** [comment]
Review status: 0 of 30 files reviewed at latest revision, 3 unresolved discussions, some commit checks failed.

---

*[mungegithub/github/github.go, line 511 \[r1\]](https://reviewable.kubernetes.io:443/reviews/kubernetes/contrib/1645#-KQT0tYG4L4EREz4SnOV:-KQT0tYG4L4EREz4SnOW:b-alza3n) ([raw file](https://github.com/kubernetes/contrib/blob/232f7654dc5a2fab0851e21235aad0775af14e48/mungegithub/github/github.go#L511)):*
> ```Go
> 	config.analytics.ListMilestones.Call(config, resp)
> 	if err != nil {
> 		glog.Errorf("Error getting milestones of state %q: %v", state, err)
> ```

Return (nil, false) here?

---

*[mungegithub/github/github.go, line 872 \[r1\]](https://reviewable.kubernetes.io:443/reviews/kubernetes/contrib/1645#-KQT1jbmEk9KTsjGJZnF:-KQT1jbmEk9KTsjGJZnG:bkkshzb) ([raw file](https://github.com/kubernetes/contrib/blob/232f7654dc5a2fab0851e21235aad0775af14e48/mungegithub/github/github.go#L872)):*
> ```Go
> 		return "", false
> 	}
> 	if !releaseMilestoneRE.MatchString(*title) {
> ```

We return true because this isn't an error condition?

---


*Comments from [Reviewable](https://reviewable.kubernetes.io:443/reviews/kubernetes/contrib/1645)*

**Comment by foxish** [comment]
@eparis I made one pass over it. I am a little worried about the fact that it will no longer be possible for any munger to see the error and act differently based on what kind of error it is, because we will always return a boolean for "something went wrong".

**Comment by grodrigues3** [comment]
[APPROVALNOTIFIER] The Following OWNERS Files Need Approval:

- **mungegithub**

We suggest the following people:
cc @bprashanth 
 You can indicate your approval by writing `/approve` in a comment
 You can cancel your approval by writing `/approve cancel`in a comment

**Comment by eparis** [comment]
After about 5 large rebases I'm thinking about just self-LGTM. We know it fixes bugs. If we want to convert to an error that's fine, but no doubt this is 100% better than it is today.

Also I intentionally used my personal e-mail on the commit to make sure the merge robot can handle cla: human-approved after we turned on protected branches.

**Comment by foxish** [comment]
@eparis Let's go ahead and merge this :)

**Pull request merged**